// @insforge/shared-schemas/dist/memory-api.schema.js
import { z } from 'zod';
// Kinds of durable agent memory.
export const memoryKindSchema = z.enum(['fact', 'decision', 'preference', 'reference']);
// POST /api/memory/remember
// Either store a single explicit memory ({ title, content }) or extract many
// durable memories from a raw task transcript ({ transcript }).
export const rememberRequestSchema = z
    .object({
    scope: z.string().min(1).default('default'),
    source: z.string().optional(),
    // explicit single-memory form
    kind: memoryKindSchema.optional(),
    title: z.string().min(1).max(2000).optional(),
    content: z.string().min(1).max(20000).optional(),
    // transcript form — capped to bound extraction LLM cost on huge inputs
    transcript: z.string().min(1).max(50000).optional(),
})
    .refine((v) => Boolean(v.transcript) || Boolean(v.title && v.content), {
    message: 'Provide either { transcript } or { title, content }',
});
export const reconcileActionSchema = z.enum(['ADD', 'UPDATE', 'NOOP']);
export const rememberResultSchema = z.object({
    action: reconcileActionSchema,
    id: z.string().uuid().optional(),
    title: z.string(),
});
export const rememberResponseSchema = z.object({
    results: z.array(rememberResultSchema),
});
// POST /api/memory/recall
export const recallRequestSchema = z.object({
    scope: z.string().min(1).default('default'),
    // capped — the query is embedded, so an unbounded string burns embedding tokens
    query: z.string().min(1).max(4000),
    limit: z.number().int().positive().max(50).default(5),
    threshold: z.number().min(0).max(1).optional(),
});
export const recalledMemorySchema = z.object({
    id: z.string().uuid(),
    kind: memoryKindSchema,
    title: z.string(),
    content: z.string(),
    similarity: z.number(),
    updated_at: z.string(),
});
export const recallResponseSchema = z.object({
    memories: z.array(recalledMemorySchema),
});
// POST /api/memory/index — cheap title-only listing (the always-load tier)
export const memoryIndexRequestSchema = z.object({
    scope: z.string().min(1).default('default'),
});
export const memoryIndexEntrySchema = z.object({
    id: z.string().uuid(),
    kind: memoryKindSchema,
    title: z.string(),
    updated_at: z.string(),
});
export const memoryIndexResponseSchema = z.object({
    entries: z.array(memoryIndexEntrySchema),
});
